src/views/Board.vue

```diff
@@ -21,23 +21,19 @@ export default {
       required: true,
     },
   },
-  computed: {
-    individuals_with_ranking() {
-      const result = [];
-
-      let i = 0;
-
-      while (i < this.individuals.length) {
-        result.push({
-          rank: i + 1,
-          username: this.individuals[i].username,
-          blocksCompleted: this.individuals[i].blocksCompleted,
-        });
+  methods: {
+    addRanking() {
+      let i = 1;
+      this.individuals.forEach((element) => {
```

Comment:
Great work, but you can actually take this a step further: 

`this.individuals.forEach((element, index) => {`
This will work the same way, and automatically manage the index for you.

I'm not super familiar with forEach, I generally use map. I assume this just modifies the original array? We might want to be creating a copy instead of modifying but that's very nitpicky, what do you think @bensenescu ?

Reply:
So here, you don't want to reassign individuals via a map because individuals is a prop. Changing a prop causes all kinds of issues. You're actually kind of changing the prop here by changing an individual in individuals, the only reason vue isn't showing errors is because it can't tell its being changed because your manipulating elements in the prop rather than the prop itself. From what I can see here, you're just displaying things in this component. So, the "best" way to do this would be to make a computed property that works like this.

computed: {
  individualsWithRank() {
    return this.individuals.map((element) => {
           return {
              username: el.username,
              blocksCompeted,
              // do your rank calculations
           }
        }
    }
}

This means you aren't manipulating a prop and then in your template, you just use the computed property as opposed to the prop.

Reply:
Another concern: @liammoyn, should the rank be returned by the backend? that seems appropriate

Reply:
Another concern: @liammoyn, should the rank be returned by the backend? that seems appropriate

Reply:
We're already returning the number of blocks completed which is essentially the rank. No need for an extra field